### PR TITLE
feat: Add prometheus metrics configuration 

### PR DESCRIPTION
**Pull request by pcantea**
* Adds prometheus metrics configuration to deployment and service
* Update readme

### CODE REVIEW

File: tutorials/helm_chart/demo/templates/servicemonitor.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{{- if and .Values.prometheus.enabled .Values.prometheus.serviceMonitor.enabled }}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: {{ .Release.Name }}
+{{- if .Values.prometheus.serviceMonitor.namespace }}
+  namespace: {{ .Values.prometheus.serviceMonitor.namespace }}
```
Comment (askore) [inline tutorials/helm_chart/demo/templates/servicemonitor.yaml:7]: Do we need a manifest to create the serviceMonitor namespace?

File: tutorials/helm_chart/demo/templates/deployment.yaml [hunk 2]
```diff
@@ -22,12 +28,15 @@ spec:
         args:
         - --model-repository=octoml/models
         - --strict-model-config=false
+        - --metrics-port={{ .Values.prometheus.metricsPort }}
```
Comment (askore) [inline tutorials/helm_chart/demo/templates/deployment.yaml:31]: I missed this, it should be added for the other ports too

**Review by askore — COMMENTED** [review]
Looking good, will test soon

File: tutorials/helm_chart/demo/templates/servicemonitor.yaml (hunk 1 above)
Comment (pcantea) [reply tutorials/helm_chart/demo/templates/servicemonitor.yaml:7]: No, the assumption is all that would be created by the user beforehand

Comment (askore) [reply tutorials/helm_chart/demo/templates/servicemonitor.yaml:7]: I think that should be mentioned in the readme, then. I would not personally assume that I had to create the namespace first.

Comment (nibalizer) [reply tutorials/helm_chart/demo/templates/servicemonitor.yaml:7]: Does it error out if the namespace doesn't exist ?

Comment (pcantea) [reply tutorials/helm_chart/demo/templates/servicemonitor.yaml:7]: It'll fail to deploy

Comment (pcantea) [reply tutorials/helm_chart/demo/templates/servicemonitor.yaml:7]: Forgot to come back to this Friday, but I have a little more clarification about the namespace here. This value sets the namespace for the serviceMonitor. Some users will have the ServiceMonitor in the same namespace as Prometheus, some will keep it with the deployment itself, those are the two generally established patterns. I don't know if we need to expand the docs, I would assume someone who knows how to run prometheus operator knows how to specify where to create the ServiceMonitor also

Comment (askore) [reply tutorials/helm_chart/demo/templates/servicemonitor.yaml:7]: This helm chart, and the whole repo, are meant as tutorials, so I don't think there is any good reason to assume such specific prior knowledge as this. If we're trying to make this helm chart production-ready, then maybe that's a more reasonable assumption, but part of making it production-ready would be fully documenting all of the variables used by the helm chart, in which case this field would still need to be documented. I think it should be added to the readme.

**Review by askore — APPROVED** [review]

**Pull request merged**